pkg/sink/sink_test.go

```diff
@@ -46,41 +43,92 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	fakedynamic "k8s.io/client-go/dynamic/fake"
-	fakekubeclientset "k8s.io/client-go/kubernetes/fake"
 	ktesting "k8s.io/client-go/testing"
+	rtesting "knative.dev/pkg/reconciler/testing"
 )
 
 const (
 	resourceLabel = triggersv1.GroupName + triggersv1.EventListenerLabelKey
 	triggerLabel  = triggersv1.GroupName + triggersv1.TriggerLabelKey
 	eventIDLabel  = triggersv1.GroupName + triggersv1.EventIDLabelKey
 
-	eventID = "12345"
+	eventID   = "12345"
+	namespace = "foo"
 )
 
 func init() {
 	// Override UID generator for consistent test results.
 	template.UID = func() string { return eventID }
 }
 
-func waitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
-	c := make(chan struct{})
-	go func() {
-		defer close(c)
-		wg.Wait()
-	}()
-	select {
-	case <-c:
-		return false
-	case <-time.After(timeout):
-		return true
+// Compare two PipelineResources for sorting purposes
+func comparePR(x, y pipelinev1.PipelineResource) bool {
+	return x.GetName() < y.GetName()
+}
+
+// getSinkAssets seeds test resources and returns a testable Sink a dynamic client.
+// The returned client is used to creating the fake resources and can be used
+// to check if the correct resources were created.
+func getSinkAssets(t *testing.T, resources test.Resources, elName string) (Sink, *fakedynamic.FakeDynamicClient) {
+	t.Helper()
+	ctx, _ := rtesting.SetupFakeContext(t)
+	clients := test.SeedResources(t, ctx, resources)
+	test.AddTektonResources(clients.Kube)
+
+	logger, _ := logging.NewLogger("", "")
+	dynamicClient := fakedynamic.NewSimpleDynamicClient(runtime.NewScheme())
+	dynamicSet := dynamicclientset.New(tekton.WithClient(dynamicClient))
+
+	r := Sink{
+		EventListenerName:      elName,
+		EventListenerNamespace: namespace,
+		DynamicClient:          dynamicSet,
+		DiscoveryClient:        clients.Kube.Discovery(),
+		KubeClientSet:          clients.Kube,
+		TriggersClient:         clients.Triggers,
+		Logger:                 logger,
+	}
+	return r, dynamicClient
+}
+
+// getCreatedPipelineResources returns the pipeline resources that was created from the given actions
+func getCreatedPipelineResources(t *testing.T, actions []ktesting.Action) []pipelinev1.PipelineResource {
+	t.Helper()
+	prs := []pipelinev1.PipelineResource{}
+	for i := range actions {
+		obj := actions[i].(ktesting.CreateAction).GetObject()
+		// Since we use dynamic client, we cannot directly get the concrete type
+		uns := obj.(*unstructured.Unstructured).Object
+		pr := pipelinev1.PipelineResource{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(uns, &pr); err != nil {
+			t.Errorf("failed to get created pipeline resource: %v", err)
+		}
+		prs = append(prs, pr)
+	}
+	return prs
+}
+
+func checkSinkResponse(t *testing.T, resp *http.Response, elName string) {
+	t.Helper()
+	if resp.StatusCode != http.StatusCreated {
+		t.Fatalf("expected response code 20 but got: %v", resp.Status)
```

Comment:
nit: `...code 20 but...` -> `...code 201 but...` or even better, we could use formatting with `http.StatusCreated`

Reply:
done!